### PR TITLE
Ge04

### PR DESCRIPTION
**Pull request by riaa-odoo**
Added 'Add Discount' button to the Sales Order form view. Clicking this will apply a -$2,500 surcharge in the motorcycle price.

### CODE REVIEW

File: ge04_team01/data/pricelist_data.xml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+        </record>
+        <record id="item_new_customers" model="product.pricelist.item">
+            <field name="pricelist_id" ref="list_new_customers" />
+            <field name="compute_price">formula</field>
+            <field name="base">list_price</field>
+            <field name="applied_on">3_global</field>
+            <field name="price_surcharge">-2500</field>
+        </record>
+    </data>
+</odoo>
```
Comment (gecm-odoo) [inline ge04_team01/data/pricelist_data.xml:16]: Missing new line in here

File: ge04_team01/models/res_partner.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+    is_new_customer = fields.Boolean("Is New Customer", compute="_compute_is_new_customer")
+
+    @api.depends("sale_order_ids")
+    def _compute_is_new_customer(self):
+        domain = [("product_id.detailed_type", "=", "motorcycle")]
+        old_customers = self.env["sale.order.line"].search(domain).mapped(
+            "order_id").filtered(lambda s: s.state in ["sale", "done"]).mapped("partner_id")
+        self.is_new_customer = True
+        for partner in old_customers:
```
Comment (gecm-odoo) [inline ge04_team01/models/res_partner.py:15]: If old_customers is a recordset of res.partner, you can avoid the `for loop` and just do something like 

old_customers.is_new_customer = False 
or
old_customers.write({'is_new_customer': False})

File: ge04_team01/views/sale_order_inherit.xml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+        <field name="inherit_id" ref="sale.view_order_form" />
+        <field name="arch" type="xml">
+            <xpath expr="//header" position="inside">
+                <field name="is_new_customer" invisible="1" />
+                <button name="apply_discount" string="Apply Discount" type="object"
+                    attrs="{'invisible': ['|', ('is_new_customer', '=', False), ('state', '!=', 'draft')]}" />
+            </xpath>
+        </field>
+    </record>
+</odoo>
```
Comment (gecm-odoo) [inline ge04_team01/views/sale_order_inherit.xml:15]: Missing new line in here as well

File: ge04_team01/models/sale_order.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
+from odoo import models, fields
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    is_new_customer = fields.Boolean(related="partner_id.is_new_customer")
```
Comment (gecm-odoo) [inline ge04_team01/models/sale_order.py:7]: I don't think you need this field.

File: ge04_team01/models/sale_order.py [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+    def apply_discount(self):
+        for sale_order in self:
```
Comment (gecm-odoo) [inline ge04_team01/models/sale_order.py:10]: `for order in self` is widely used on standard. It is not incorrect what you did but the standard way is the preferred way.

File: ge04_team01/models/sale_order.py [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+            if sale_order.is_new_customer:
```
Comment (gecm-odoo) [inline ge04_team01/models/sale_order.py:11]: if order.partner_id.is_new_customer

**Review by gecm-odoo — CHANGES_REQUESTED** [review]
Good job! Some small changes and it should be ready to push

File: ge04_team01/models/sale_order.py (hunk 4 above)
Comment (riaa-odoo) [reply ge04_team01/models/sale_order.py:7]: I think it's necessary because otherwise we can't add it into the Sale Order form viev.

Comment (gecm-odoo) [reply ge04_team01/models/sale_order.py:7]: Sorry, did not realize you need the field on the view

**Review by gecm-odoo — APPROVED** [review]
Great job!

**Pull request merged**